### PR TITLE
Migrate to applications service client

### PR DESCRIPTION
**Pull request by spraxxy**
Removed references to current system of Application and ApplicationConfiguration and replaced them with ApplicationsServiceClient

### CODE REVIEW

File: src/main/java/org/atlasapi/application/ApplicationModule.java [hunk 1]
```diff
@@ -1,35 +1,28 @@
...
-	public @Bean ApplicationConfigurationFetcher configFetcher(){
-		return new IpCheckingApiKeyConfigurationFetcher(applicationStore());
-	}
-	
-	public @Bean ApplicationStore applicationStore(){
-	    IdGenerator idGenerator = new MongoSequentialIdGenerator(mongo, "application");
-	    return new MongoApplicationStore(mongo, idGenerator);
-	}
-}
+    private final String host = Configurer.get("thing").get(); //TODO: fix
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/ApplicationModule.java]: Yes, fix ;)

File: src/main/java/org/atlasapi/application/ApplicationModule.java [hunk 2]
```diff
@@ -1,35 +1,28 @@
...
+	@Bean
+	public ApplicationFetcher configFetcher(){
+        return new ApiKeyApplicationFetcher(applicationsClient());
+    }
+
+    @Bean
+    public ApplicationsClient applicationsClient(){
+        return ApplicationsClientImpl.create(host , new MetricRegistry());
+    }
+}
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/ApplicationModule.java]: Newline at end of file

File: src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import com.metabroadcast.applications.client.model.internal.Environment;
+import com.metabroadcast.applications.client.query.Query;
+import com.metabroadcast.applications.client.query.Result;
+import com.metabroadcast.common.properties.Configurer;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ApiKeyApplicationFetcher implements ApplicationFetcher {
+
+    public static final String API_KEY_QUERY_PARAMETER = "apiKey";
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java]: This doesn't need to be public

File: src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    private final Environment environment;
+    
+    private final ApplicationsClient applicationsClient;
+
+    public ApiKeyApplicationFetcher(ApplicationsClient applicationsClient) {
+        this.applicationsClient = applicationsClient;
+        this.environment = checkNotNull(Environment.parse(Configurer.getPlatform()));
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java]: Pass the environment in as a constructor parameter.

File: src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ApiKeyApplicationFetcher implements ApplicationFetcher {
+
+    public static final String API_KEY_QUERY_PARAMETER = "apiKey";
+    private final Environment environment;
+    
+    private final ApplicationsClient applicationsClient;
+
+    public ApiKeyApplicationFetcher(ApplicationsClient applicationsClient) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java]: Code style. Private constructor + static factory method

File: src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        this.applicationsClient = applicationsClient;
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java]: checkNotNull

File: src/main/java/org/atlasapi/application/query/ApplicationFetchException.java [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
 
+import com.metabroadcast.applications.client.exceptions.ErrorCode;
 
 public abstract class ApplicationFetchException extends Exception {
     private static final long serialVersionUID = 5835729721346196714L;
     private final String apiKey;
+    private final ErrorCode errorCode;
 
-    public ApplicationFetchException(String apiKey) {
+    public ApplicationFetchException(String apiKey, ErrorCode errorCode) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/ApplicationFetchException.java]: The class is abstract. Constructor should be `protected`

File: src/main/java/org/atlasapi/application/query/InvalidApiKeyException.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package org.atlasapi.application.query;
+
+import com.metabroadcast.applications.client.exceptions.ErrorCode;
+
+public class InvalidApiKeyException extends ApplicationFetchException {
+    private static final long serialVersionUID = -8204400513571208163L;
+    
+    public InvalidApiKeyException(String apiKey, ErrorCode errorCode) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/query/InvalidApiKeyException.java]: Code style. Private constructor + static factory method

File: src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java (hunk 3 above)
Comment (jamiesprax) [inline src/main/java/org/atlasapi/application/query/ApiKeyApplicationFetcher.java]: This is used by SearchController:57

**Pull request closed without merging**